test/suite/test_timestamp03.py

```diff
@@ -215,27 +270,51 @@ def test_timestamp03(self):
             count += 1
         # print "Updated " + str(count) + " keys to value2"
 
+        # Scenario : 4
+        # Now check that we don't see the updated data of timestamp tables
+        # with old read_timestamp
+        self.check(self.session, 'read_timestamp=' + timestamp_str(100),
+            self.tablename, dict((k, self.value) for k in orig_keys))
+        self.check(self.session, 'read_timestamp=' + timestamp_str(100),
+            self.tablename2, dict((k, self.value2) for k in orig_keys))
+        self.check(self.session, 'read_timestamp=' + timestamp_str(100),
+            self.tablename3, dict((k, self.value) for k in orig_keys))
+        self.check(self.session, 'read_timestamp=' + timestamp_str(100),
+            self.tablename4, dict((k, self.value2) for k in orig_keys))
+
+        # Scenario : 5
+        # Now check that we see the expected state when reading at each
+        # updated timestamp
+        valdict = dict((k, self.value) for k in orig_keys)
+        for i, t in enumerate(orig_keys):
+            valdict[i+1] = self.value2
+            self.check(self.session, 'read_timestamp=' + timestamp_str(t + 100),
+                self.tablename, valdict)
+            self.check(self.session, 'read_timestamp=' + timestamp_str(t + 100),
+                self.tablename2, dict((k, self.value2) for k in orig_keys))
+            self.check(self.session, 'read_timestamp=' + timestamp_str(t + 100),
+                self.tablename3, valdict)
+            self.check(self.session, 'read_timestamp=' + timestamp_str(t + 100),
+                self.tablename4, dict((k, self.value2) for k in orig_keys))
+
         # Take a checkpoint using the given configuration.  Then verify
         # whether value2 appears in a copy of that data or not.
-        valcnt2 = valcnt3 = nkeys
+        valcnt2 = valcnt3 = valcnt4 = nkeys
         if self.val == 'all':
             valcnt = nkeys
         else:
             valcnt = 0
-        self.ckpt_backup(self.value2, valcnt, valcnt2, valcnt3)
+        self.ckpt_backup(self.value2, valcnt, valcnt2, valcnt3, valcnt4, "")
         if self.ckptcfg != 'read_timestamp':
             # Update the stable timestamp to the latest, but not the oldest
             # timestamp and make sure we can see the data.  Once the stable
             # timestamp is moved we should see all keys with value2.
             self.conn.set_timestamp('stable_timestamp=' + \
                 timestamp_str(100+nkeys))
-            self.ckpt_backup(self.value2, nkeys, nkeys, nkeys)
-
-        # If we're not using the log we're done.
-        if not self.using_log:
-            return
+            self.ckpt_backup(self.value2, nkeys, nkeys, nkeys, nkeys, "")
 
-        # Update them and retry.  This time take a backup and recover.
+        # Update the keys and checkpoing using stable_timestamp, not using
```

Comment:
checkpoing => checkpoint.
I think most or all of the original comments talk about the timestamps in normal words rather than config strings.  So please check in all the comments for that.  I.e. here I suggest:
`# Update the keys and checkpoint using the stable timestamp.`